commands/provider/service_provider.go

```diff
@@ -94,6 +94,10 @@ func (p *ServiceProvider) ServiceListenEvents(id, eventFilter string) (chan *cor
 	resultC := make(chan *coreapi.EventData)
 	errC := make(chan error)
 
+	go func() {
+		<-stream.Context().Done()
+		errC <- stream.Context().Err()
```

Comment:
What if there is an error in the context but it doesn't close ? just want to make sure that it's not a problem here.

What about something like that ?

```go
go func() {
  select {
    case <- stream.Context().Done()
      errC <- fmt.Errorf("context closed")
    case err := <- stream.Context().Error()
      errC <- err
  }
}
```

Reply:
`stream.Context().Error()` is not a channel.
The way is done is the right way to do.

Reply:
sorry yes my bad... that's friday night that's why...